### PR TITLE
Avoid evaluating kernel in `expand_batch`

### PR DESCRIPTION
**Pull request by dannyfriar**
This is a very similar case to #2140.

Not necessarily expecting this to be merged as is but just curious to see if anything breaks with this tiny change. This also results in materialising the NxN matrix in some cases.

### CODE REVIEW

**Review by gpleiss — APPROVED** [review]
LGTM. I'll wait till the `add_jitter` PR is merged before merging this.

**Comment by dannyfriar** [comment]
@gpleiss what is the expected release cadence for GPyTorch? I just wanted to get an idea of when I can expect this to be available in pypi.

Thanks!

**Pull request merged**
